flake.nix

```diff
@@ -0,0 +1,124 @@
+{
+  description = "Nix package for centrifuge-chain";
+
+  inputs = {
+    nixpkgs.url = github:NixOS/nixpkgs/nixos-21.11;
+    gitignore = {
+      url = github:hercules-ci/gitignore.nix;
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+    fenix = {
+      url = github:nix-community/fenix;
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+
+  };
+
+  outputs = inputs:
+    let
+      name = "centrifuge-chain";
+      major = "0.10.9";
+      version = "${major}-${commit-substr}";
+
+      # This evaluates to the first 6 digits of the git hash of this repo's HEAD
+      # commit, or to "dirty" if there are uncommitted changes.
+      commit-substr = builtins.substring 0 6 (inputs.self.rev or "dirty");
```

Comment:
The git short is usually 7 digits, not sure if `builtins.substring` is upperbound inclusive or not, but would love to keep using 7.

Reply:
In addition, since we currently add a time component to the image (easier to filter by), can we keep the same here?

Reply:
What do you mean with time component? The date the image was created? If so, that's possible, but it makes the image impure (since it depends on creation date), and therefore not 100% reproducible.

Reply:
Changed hash length to 7.

Reply:
I've also pinned nightly to the version 2021-11-07.

Reply:
@mikiquantum I've added a commit that fixes the image creation date. Compare the first and the second line:

```
❯ podman images
REPOSITORY                            TAG         IMAGE ID      CREATED       SIZE
localhost/centrifugeio/go-centrifuge  2.0-dirty   c3a8eb0b74ce  3 days ago    70.5 MB
localhost/centrifugeio/go-centrifuge  2.0-5ae4cd  829162b1650d  52 years ago  70.5 MB
```